src/main/java/org/owasp/benchmark/score/parsers/HdivReader.java

```diff
@@ -0,0 +1,162 @@
+package org.owasp.benchmark.score.parsers;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.owasp.benchmark.score.BenchmarkScore;
+
+public class HdivReader extends Reader {
+
+	Set<String> invalid = new HashSet<>();
+
+	public static void main(final String[] args) throws Exception {
+		File f = new File("hdivAgentLog.hlg");
+		HdivReader cr = new HdivReader();
+		System.out.println(cr.parse(f));
+	}
+
+	public TestResults parse(final File f) throws Exception {
+		TestResults tr = new TestResults("Hdiv", true, TestResults.ToolType.IAST);
+
+		BufferedReader reader = new BufferedReader(new FileReader(f));
+		String firstLine = reader.readLine();
+		String lastLine = "";
+		String line = "";
+		ArrayList<String> chunk = new ArrayList<>();
+		String testNumber = "00001";
+		while (line != null) {
+			try {
+				line = reader.readLine();
+				if (line != null) {
+					if (line.contains("SourceCodeVulnerability [")) {
+						// ok, we're starting a new URL, so process this one and start the next chunk
+						process(tr, testNumber, Arrays.asList(line));
+						chunk.clear();
+						testNumber = "00000";
+						String fname = "/" + BenchmarkScore.BENCHMARKTESTNAME;
+						int idx = line.indexOf(fname);
+						if (idx != -1) {
+							testNumber = line.substring(idx + fname.length(), idx + fname.length() + 5);
+						}
+						lastLine = line;
+					}
+					else if (line.contains("Product version:")) {
+						String version = line.substring(line.indexOf("version:")+"version:".length());
+						tr.setToolVersion(version);
+					}
+				}
+			}
+			catch (Exception ex) {
+				ex.printStackTrace();
+			}
+		}
+		reader.close();
+		tr.setTime(calculateTime(firstLine, lastLine));
+		return tr;
+	}
+
+	private String calculateTime(final String firstLine, final String lastLine) {
+		try {
+			try {
+				return calculateTime(firstLine, lastLine, 0);
+			} catch (ParseException e) {
+				return calculateTime(firstLine, lastLine, 1);
+			}
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+
+	private String calculateTime(final String firstLine, final String lastLine, final int timeColumn) throws ParseException {
+		String start = firstLine.split(" ")[timeColumn];
+		String stop = lastLine.split(" ")[timeColumn];
+		SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss,SSS");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.